src/main/reactjs/src/redux/selectors/clerkTranslator.ts

```diff
@@ -87,26 +122,104 @@ const isAuthorisationExpiringSoon = (
 
 const filterByAuthorisationStatus = (
   translator: ClerkTranslator,
-  status: AuthorisationStatus
+  status: AuthorisationStatus,
+  currentDate: Date
 ) => {
-  const now = new Date();
-  switch (status) {
+  const expiringSoonDate = expiringSoonTreshold(currentDate);
+
+  return translator.authorisations.find((a) =>
+    matchesAuthorisationStatus(
+      { authorisationStatus: status },
+      currentDate,
+      expiringSoonDate,
+      a
+    )
+  );
+};
+
+const filterByAuthorisationCriteria = (
+  translator: ClerkTranslator,
+  filters: ClerkTranslatorFilter,
+  currentDate: Date
+) => {
+  const expiringSoonDate = expiringSoonTreshold(currentDate);
+
+  return translator.authorisations.find(
+    (a) =>
+      matchesFromLang(filters, a) &&
+      matchesToLang(filters, a) &&
+      matchesAuthorisationBasis(filters, a) &&
+      matchesAuthorisationStatus(filters, currentDate, expiringSoonDate, a)
+  );
+};
+
+const matchesFromLang = (
+  { fromLang }: ClerkTranslatorFilter,
+  authorisation: Authorisation
+) => {
+  if (fromLang) {
+    return authorisation.languagePair.from == fromLang;
+  }
+
+  return true;
+};
+
+const matchesToLang = (
+  { toLang }: ClerkTranslatorFilter,
+  authorisation: Authorisation
+) => {
+  if (toLang) {
+    return authorisation.languagePair.to == toLang;
+  }
+
+  return true;
+};
+
+const matchesAuthorisationBasis = (
+  { authorisationBasis }: ClerkTranslatorFilter,
+  authorisation: Authorisation
+) => {
+  if (authorisationBasis) {
+    return authorisation.basis == authorisationBasis;
+  }
+
+  return true;
+};
+
+const matchesAuthorisationStatus = (
+  { authorisationStatus }: ClerkTranslatorFilter,
+  currentDate: Date,
+  expiringSoonThreshold: Date,
```

Comment:
This value can be calculated under this method and doesn't have to be passed as a parameter.

Reply:
Could, but I didn't want to construct new Date instances for each translator. Not sure if it's a big deal from eg. a performance point-of-view. I do think it's more consistent this way, though.